synapse/lib/stormlib/cortex.py

```diff
@@ -0,0 +1,839 @@
+        })
+
+    def getObjLocals(self):
+        return {
+            'pack': self._methPack,
+        }
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _methPack(self):
+        return copy.deepcopy(self.info)
+
+    @s_stormtypes.stormfunc(readonly=True)
+    def _ctorMethods(self, path=None):
+        return HttpApiMethods(self)
+
+    async def _storPath(self, path):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        path = await s_stormtypes.tostr(path)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'path', path)
+        self.info['path'] = path
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorPath(self):
+        return self.info.get('path')
+
+    async def _storName(self, name):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        name = await s_stormtypes.tostr(name)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'name', name)
+        self.info['name'] = name
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorView(self):
+        iden = self.info.get('view')
+        vdef = await self.runt.snap.core.getViewDef(iden)
+        if vdef is None:
+            raise s_exc.NoSuchView(mesg=f'No view with {iden=}', iden=iden)
+        return s_stormtypes.View(self.runt, vdef, path=self.path)
+
+    async def _storVars(self, varz):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        varz = await s_stormtypes.toprim(varz)
+        adef = await self.runt.snap.core.modHttpExtApi(self.iden, 'vars', varz)
+        _varz = self.info.get('vars')
+        _varz.clear()
+        _varz.update(**adef.get('vars'))
+
+    async def _storView(self, iden):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        if isinstance(iden, s_stormtypes.View):
+            view = iden.value().get('iden')
+        else:
+            view = await s_stormtypes.tostr(iden)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'view', view)
+        self.info['view'] = view
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorName(self):
+        return self.info.get('name')
+
+    async def _storDesc(self, desc):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        desc = await s_stormtypes.tostr(desc)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'desc', desc)
+        self.info['desc'] = desc
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorDesc(self):
+        return self.info.get('desc')
+
+    async def _storRunas(self, runas):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        runas = await s_stormtypes.tostr(runas)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'runas', runas)
+        self.info['runas'] = runas
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorRunas(self):
+        return self.info.get('runas')
```

Comment:
should this grab a heavy user object like owner does?

Reply:
nevermind, forgot this was just a string.

Reply:
No since this is just a string that is either `user` or `owner`